spec/beaker/dsl/install_utils/foss_utils_spec.rb

```diff
@@ -877,6 +877,25 @@ def stub_uninteresting_portions_of_install_puppetlabs_dev_repo!
       subject.install_puppet_agent_dev_repo_on( host, opts )
     end
 
+    it 'runs the correct install for osx platforms (newest link format)' do
+      platform = Object.new()
+      allow(platform).to receive(:to_array) { ['osx', '10.9', 'x86_64', 'mavericks']}
+      host = basic_hosts.first
+      host['platform'] = platform
+      opts = { :version => '0.1.0' }
+      allow( subject ).to receive( :options ).and_return( {} )
+      copied_path = "#{win_temp}\\puppet-agent-x64.msi"
+      mock_echo = Object.new()
+      allow( mock_echo ).to receive( :raw_output ).and_return( copied_path )
+
+      expect(subject).to receive(:link_exists?).with(/\/puppet-agent\/0.1.0\/repos\/apple\/10.9\/PC1\/x86_64\//).and_return(true).twice
```

Comment:
This might be more readable if something like `/#{Regexp.escape('puppet-agent/0.1.0/repos/apple/10.9/PC1/x86_64')}/` were used instead of escaping the above.